benchmarks_test.go

```diff
@@ -0,0 +1,235 @@
+	sndr, err := session.NewSender(ctx, "target", &SenderOptions{
+		SettlementMode: SenderSettleModeUnsettled.Ptr(),
+	})
+	cancel()
+	require.NoError(b, err)
+	sendInitialFlowFrame(b, conn, 0, 1000000)
+	b.ResetTimer()
+	b.ReportAllocs()
+
+	msg := NewMessage([]byte("test"))
+	for i := 0; i < b.N; i++ {
+		ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+		err = sndr.Send(ctx, msg, nil)
+		cancel()
+		require.NoError(b, err)
+	}
+}
+
+func BenchmarkSenderSendSSMSettled(b *testing.B) {
+	responder := func(req frames.FrameBody) ([]byte, error) {
+		b, err := senderFrameHandler(SenderSettleModeSettled)(req)
+		if b != nil || err != nil {
+			return b, err
+		}
+		switch req.(type) {
+		case *frames.PerformFlow:
+			return nil, nil
+		case *frames.PerformDisposition:
+			return nil, nil
+		case *frames.PerformTransfer:
+			return nil, nil
+		default:
+			return nil, fmt.Errorf("unhandled frame %T", req)
+		}
+	}
+	conn := fake.NewNetConn(responder)
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	client, err := NewConn(ctx, conn, nil)
+	cancel()
+	require.NoError(b, err)
+	ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+	session, err := client.NewSession(ctx, nil)
+	cancel()
+	require.NoError(b, err)
+	ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+	sndr, err := session.NewSender(ctx, "target", &SenderOptions{
+		SettlementMode: SenderSettleModeSettled.Ptr(),
+	})
+	cancel()
+	require.NoError(b, err)
+	sendInitialFlowFrame(b, conn, 0, 1000000)
+	b.ResetTimer()
+	b.ReportAllocs()
+
+	msg := NewMessage([]byte("test"))
+	for i := 0; i < b.N; i++ {
+		ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+		err = sndr.Send(ctx, msg, nil)
+		cancel()
+		require.NoError(b, err)
+	}
+}
+
+func BenchmarkReceiverReceiveRSMFirst(b *testing.B) {
+	responder := func(req frames.FrameBody) ([]byte, error) {
+		b, err := receiverFrameHandler(ReceiverSettleModeFirst)(req)
+		if b != nil || err != nil {
+			return b, err
+		}
+		switch req.(type) {
+		case *frames.PerformFlow:
+			return nil, nil
+		case *frames.PerformDisposition:
+			return nil, nil
+		default:
+			return nil, fmt.Errorf("unhandled frame %T", req)
+		}
+	}
+	conn := fake.NewNetConn(responder)
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	client, err := NewConn(ctx, conn, nil)
+	cancel()
+	require.NoError(b, err)
+	ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+	session, err := client.NewSession(ctx, nil)
+	cancel()
+	require.NoError(b, err)
+	ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+	rcvr, err := session.NewReceiver(ctx, "source", &ReceiverOptions{
+		SettlementMode: ReceiverSettleModeFirst.Ptr(),
+	})
+	cancel()
+	require.NoError(b, err)
+	b.ResetTimer()
+	b.ReportAllocs()
+
+	for i := 0; i < b.N; i++ {
+		b.StopTimer()
+		fr, err := fake.PerformTransfer(0, 0, uint32(i), []byte{})
```

Comment:
I'm going to see if I can remove this from the loop to avoid including these allocations in the results.

Reply:
Done.